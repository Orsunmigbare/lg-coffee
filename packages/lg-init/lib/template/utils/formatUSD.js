/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  const string = JSON.stringify(number);
  const { length } = string;
  return JSON.parse(
    `${string.substring(0, length - 2)}.${string.substring(length - 2)}`
  ).toFixed(2);
};

export { formatUSD };
