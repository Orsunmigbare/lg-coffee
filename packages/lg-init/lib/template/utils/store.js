module.exports = {
  URL: process.env.LOCAL_URL || process.env.PUBLIC_URL,
  storeName: 'VisionSpeaks',
  logo: '/logo.png',
  accountOwner: 'Yulissa V',
  products: [
    {
      name: 'Lady In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'lad_vs_',
      images: [
        '/lady_in_the_void_TRANSP.png',
        '/lady_001.png',
        '/lady_002.png',
      ],
    },
    {
      name: 'Bunns In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'bun_vs_',
      images: [
        '/bunns_in_the_void_TRANSP.png',
        '/bunns_001.png',
        '/bunns_002.png',
      ],
    },
  ],
  pages: {
    globals: {
      // navigations
      navigation: [
        {
          title: 'Products',
          path: '/',
        },
        {
          title: 'Made To Order',
          path: '/info/contact',
        },
        {
          title: 'About',
          path: '/info/about',
        },
        {
          title: 'Cart',
          path: '/view-cart',
        },
      ],
      footer: [
        {
          title: 'Contact',
          path: 'mailto:support@visionspeaks.art',
        },
        {
          title: 'FAQ',
          path: '/info/faq',
        },
      ],
    },
    homePage: {
      hero: {
        image: '/heatpress.jpeg',
        headline: 'Made To Order',
        subHeadline: 'The highest quality work comes from custom work.',
        buttonName: 'Start Your Order',
        buttonPath: '/products',
        styles: {},
      },
    },
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
      about: `
        VisionSpeaks is a custom art collective.
      `,
    },
  },
};
