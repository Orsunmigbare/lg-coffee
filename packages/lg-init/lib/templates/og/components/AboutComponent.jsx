/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import { H4, P } from './_helpers';

const AboutComponent = ({ store }) => (
  <>
    <H4>About Us</H4>
    <P>{store.about || 'Coming Soon'}</P>
  </>
);

AboutComponent.propTypes = {
  store: PropTypes.shape({
    about: PropTypes.string,
  }),
};

export default AboutComponent;
