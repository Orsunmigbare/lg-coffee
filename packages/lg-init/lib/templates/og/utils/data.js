exports.data = {
  store: {
    name: 'VisionSpeaks',
    logo: '/logo.png',
    url: '',
    description: 'Custom sticker shop',
    contact: 'youremail@here.com',
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
    },
    social: {
      facebook: '',
      twitter: '',
      instagram: '',
    },
  },
  theme: {
    text: { color: '#212529' },
    secondaryText: { color: '#6C757D' },
    primaryBackground: { backgroundColor: '#F8F9FA' },
    secondaryBackground: { backgroundColor: '#FFFFFF' },
  },
  products: {
    heading: 'New Arrivials',
    products: [
      {
        name: 'Bunns In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'bun_vs_',
        images: [
          '/bunns_in_the_void_TRANSP.png',
          '/bunns_001.png',
          '/bunns_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
      {
        name: 'Lady In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'lad_vs_',
        images: [
          '/lady_in_the_void_TRANSP.png',
          '/lady_001.png',
          '/lady_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
    ],
  },
};
