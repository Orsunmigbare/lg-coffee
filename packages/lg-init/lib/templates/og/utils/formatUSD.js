/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {     
  const { length } = number;
  const str = `${number.substring(0, length - 2)}.${number.substring(length - 2)}`;  
  return parseFloat(str).toFixed(2)
};

export { formatUSD };
