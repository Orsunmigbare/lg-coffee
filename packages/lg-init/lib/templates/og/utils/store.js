exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.png',
  hero: '/hero.png',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
