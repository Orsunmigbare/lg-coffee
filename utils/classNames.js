exports.cn = {
  darkForm: {
    formWrapper:
      'd-flex flex-column align-items-center justify-content-center w-100',
    form:
      'shadow-pop-br my-5 form-signin d-flex flex-column align-items-center w-50 border border-primary rounded p-3 py-5',
    input: 'form-control mb-2 bg-dark text-light border-0 my-2',
    label: 'sr-only',
    button: 'btn btn-outline-light align-self-end mt-2',
    h1: 'text-light align-self-start',
  },
};
