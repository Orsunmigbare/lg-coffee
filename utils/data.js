exports.data = {
  store: {
    accountOwner: "Yulissa V",
    dateCreated: "08/2020",
    storeName: "VisionSpeaks",
    logo: "/logo.png",
    hero: {
      heroImage: "/heatpress.jpeg",
      heroHeadline: "Made To Order",
      heroSubHeadline: "The highest quality work comes from custom work.",
      heroButton: "Start Your Order",
      heroPath: "/products",
      styles: {},
    },
    nav: {
      links: [
        {
          title: "Products",
          path: "/",
        },
        {
          title: "Made To Order",
          path: "/made-to-order",
        },
        {
          title: "About",
          path: "/about",
        },
      ],
      style: {
        nav: {
          background: "#FAFAFA",
        },
        link: {
          color: "#181818",
        },
      },
    },
    footer: {
      col: [
        {
          title: "Contact",
          link: "/contact",
        },
        {
          title: "Contact",
          link: "/contact",
        },
      ],
    },
  },
  products: [
    {
      imagePath: "/front_page_bunns@2x.png",
      imagePathHovered: "/front_page_bunns_hover@2x.png", // drop image opacity to .6
      productTitle: "Bunns In The Void",
      productDescription: "Holographic Sticker, This thing is bad ass",
      productPrice: "$2.87",
      productSku: "/bun_vs_",
      productBgIsLight: "true",
    },
    {
      imagePath: "/front_page_lady@2x.png",
      imagePathHovered: "/front_page_lady_hover@2x.png",
      productTitle: "Lady In The Void",
      productDescription: "Holographic Sticker, This thing is bad ass",
      productPrice: "$2.87",
      productSku: "/lad_vs_",
      productBgIsLight: false,
    },
  ],
};
