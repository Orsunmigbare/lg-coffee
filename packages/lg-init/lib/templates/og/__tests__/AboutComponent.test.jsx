import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AboutComponent from '../components/AboutComponent';
import { testData } from '../utils/testData';

it('AboutComponent component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AboutComponent store={testData.store} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('AboutComponent component renders unchanged', () => {
  const tree = renderer.create(<AboutComponent store={testData.store} />).toJSON();
  expect(tree).toMatchSnapshot();
});