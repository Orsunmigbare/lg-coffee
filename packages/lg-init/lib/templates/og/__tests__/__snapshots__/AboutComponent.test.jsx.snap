// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent component renders unchanged 1`] = `
Array [
  <h4
    className=""
  >
    About Us
  </h4>,
  <p
    className=""
  >
    Everyone loves a happy dog!
  </p>,
]
`;
